build.gradle.kts

```diff
@@ -0,0 +1,304 @@
+}
+
+/**
+ * Combines the 'coreRuntime' with the patched external module jars
+ */
+val runtime by configurations.creating {
+    isVisible = false
+    extendsFrom(coreRuntime)
+}
+
+val gradlePlugins by configurations.creating {
+    isVisible = false
+}
+
+val testRuntime by configurations.creating {
+    extendsFrom(runtime)
+    extendsFrom(gradlePlugins)
+}
+
+configurations {
+    all {
+        usage(Usage.JAVA_RUNTIME)
+    }
+}
+
+extra["allTestRuntimeDependencies"] = testRuntime.allDependencies
+
+val patchedExternalModulesDir = buildDir / "external/files"
+val patchedExternalModules = files(provider { fileTree(patchedExternalModulesDir).files.sorted() })
+patchedExternalModules.builtBy("patchExternalModules")
+
+dependencies {
+
+    externalModules("org.gradle:gradle-kotlin-dsl:${BuildEnvironment.gradleKotlinDslVersion}")
+    externalModules("org.gradle:gradle-kotlin-dsl-tooling-builders:${BuildEnvironment.gradleKotlinDslVersion}")
+
+    coreRuntime(project(":launcher"))
+    coreRuntime(project(":runtimeApiInfo"))
+
+    runtime(project(":wrapper"))
+    runtime(project(":installationBeacon"))
+    runtime(patchedExternalModules)
+
+    pluginProjects.forEach { gradlePlugins(it) }
+    implementationPluginProjects.forEach { gradlePlugins(it) }
+    gradlePlugins(project(":workers"))
+    gradlePlugins(project(":dependencyManagement"))
+    gradlePlugins(project(":testKit"))
+
+    coreRuntimeExtensions(project(":dependencyManagement")) //See: DynamicModulesClassPathProvider.GRADLE_EXTENSION_MODULES
+    coreRuntimeExtensions(project(":pluginUse"))
+    coreRuntimeExtensions(project(":workers"))
+    coreRuntimeExtensions(patchedExternalModules)
+}
+
+extra["allCoreRuntimeExtensions"] = coreRuntimeExtensions.allDependencies
+
+task<PatchExternalModules>("patchExternalModules") {
+    allModules = externalModulesRuntime
+    coreModules = coreRuntime
+    modulesToPatch = this@Build_gradle.externalModules
+    destination = patchedExternalModulesDir
+}
+
+evaluationDependsOn(":distributions")
+
+val gradle_installPath: Any? by project
+
+task<Install>("install") {
+    description = "Installs the minimal distribution into directory $gradle_installPath"
+    group = "build"
+    with(distributionImage("binDistImage"))
+    installDirPropertyName = ::gradle_installPath.name
+}
+
+task<Install>("installAll") {
+    description = "Installs the full distribution into directory $gradle_installPath"
+    group = "build"
+    with(distributionImage("allDistImage"))
+    installDirPropertyName = ::gradle_installPath.name
+}
+
+fun distributionImage(named: String) =
+    project(":distributions").property(named) as CopySpec
+
+afterEvaluate {
+    if (gradle.startParameter.isBuildCacheEnabled) {
+        rootProject
+            .availableJavaInstallations
+            .validateBuildCacheConfiguration(buildCacheConfiguration())
+    }
+}
+
+fun Project.buildCacheConfiguration() =
+    (gradle as GradleInternal).settings.buildCache
```

Comment:
:o: Same comment as before.
Seems weird you are using an internal API in your own build.